paint/PAINT_Tool.h

```diff
@@ -17,8 +19,10 @@ namespace paint
 
 		void setActiveBrush(std::shared_ptr<Brush> brush);
 		void clearLines();
-		virtual void toolFunction(win::Coords relCoords, win::Coords prevRelCoords) = 0;
+		virtual void toolFunction(win::Coords mouseCoords = { 0,0 }, win::Coords prevMouseCoords = { 0,0 }, win::Coords startCoords = { 0,0 }, gfx::Rectangle refRect = {0,0,0,0}) = 0;
 		std::shared_ptr<Brush> getActiveBrush() const {return activeBrush_;}
+		void renderLines(SDL_Renderer* renderer, const std::vector<gfx::Line>& lines) const;
+		virtual void toolFunctionEnd(win::Coords mouseCoords, win::Coords prevMouseCoords, win::Coords startCoords, gfx::Rectangle refRect) = 0;
 
 	protected:
 		std::vector<gfx::Line> lines_;
```

Comment:
This is implementation, not interface, so should be private. Protected fields rupture encapsulation. Either use a set / get interface, or change the design so as not to need this.
**MUST**

Reply:
done